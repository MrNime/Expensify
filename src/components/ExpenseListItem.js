import React from 'react';
import moment from 'moment';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({
  id, description, amount, createdAt,
}) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h4>{description}</h4>
    </Link>
    <p>{numeral(amount / 100).format('$0,0.00')}</p>
    {
      //    <p>â‚¬{(amount / 100).toFixed(2)}</p>
    }
    <p>{moment(createdAt).format('DD/MM/YYYY')}</p>
  </div>
);

export default ExpenseListItem;
