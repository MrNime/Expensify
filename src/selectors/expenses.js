import moment from 'moment';

// Get visible expenses
export default (expenses, {
  text, sortBy, sortDirection, startDate, endDate,
}) =>
  expenses
    .filter((expense) => {
      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
      const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
      const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === 'date') {
        if (sortDirection === 'descending') {
          return a.createdAt < b.createdAt ? 1 : -1;
        }
        return a.createdAt < b.createdAt ? -1 : 1;
      } else if (sortBy === 'amount') {
        if (sortDirection === 'descending') {
          return a.amount < b.amount ? 1 : -1;
        }
        return a.amount < b.amount ? -1 : 1;
      } else if (sortBy === 'description') {
        if (sortDirection === 'descending') {
          return a.description < b.description ? -1 : 1;
        }
        return a.description < b.description ? 1 : -1;
      } else if (sortBy === 'paid') {
        if (sortDirection === 'descending') {
          return a.paidAmount < b.paidAmount ? -1 : 1;
        }
        return a.paidAmount < b.paidAmount ? 1 : -1;
      } else if (sortBy === 'saved') {
        if (sortDirection === 'descending') {
          return a.amount - a.paidAmount <= b.amount - b.paidAmount ? 1 : -1;
        }
        return a.amount - a.paidAmount <= b.amount - b.paidAmount ? -1 : 1;
      }
    });
