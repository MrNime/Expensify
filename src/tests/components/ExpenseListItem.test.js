import React from 'react';
import { shallow } from 'enzyme';
import 'moment/locale/nl-be';
import numeral from 'numeral';
import 'numeral/locales/nl-be';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses.test';

beforeAll(() => {
  numeral.locale('nl-be');
});

test('should render ExpenseListItem', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
