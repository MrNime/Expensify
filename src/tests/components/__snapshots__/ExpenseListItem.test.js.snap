// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h4>
      Gum
    </h4>
  </Link>
  <p>
    $1.95
  </p>
  <p>
    01/01/1970
  </p>
</div>
`;
